Pages/TrackFood.js

```diff
@@ -69,20 +87,54 @@ export default function TrackFood() {
   );
 }
 
-const handlePress = async (db, breakfast, lunch, dinner, snacks, nav) => {
-  await db.runAsync(
-    "INSERT INTO foodLog (time, breakfast, lunch, dinner, snacks) VALUES (?, ?, ?, ?, ?);",
-    Date.now(),
-    breakfast,
-    lunch,
-    dinner,
-    snacks
-  );
+const handlePress = async (
+  db,
+  breakfast,
+  lunch,
+  dinner,
+  snacks,
+  nav,
+  firstEntryToday,
+  lastEntryTime
+) => {
+  if (firstEntryToday) {
+    await db.runAsync(
+      "INSERT INTO foodLog (time, date, breakfast, lunch, dinner, snacks) VALUES (?, ?, ?, ?, ?, ?);",
+      Date.now(),
+      new Date(Date.now()).toISOString(),
+      breakfast,
+      lunch,
+      dinner,
+      snacks
+    );
+  } else {
+    console.log("hello");
+    await db.runAsync(
+      "UPDATE foodLog SET time = ?, date = ?, breakfast = ?, lunch = ?, dinner = ?, snacks = ? WHERE time = ?;",
+      Date.now(),
+      new Date(Date.now()).toISOString(),
+      breakfast,
+      lunch,
+      dinner,
+      snacks,
+      lastEntryTime
+    );
+    console.log(lastEntryTime);
+  }
   // print table for debugging purposes
   console.log(await db.getAllAsync("SELECT * FROM foodLog"));
   nav.goBack();
 };
 
+const loadData = async (db) => {
+  const data = await db.getAllAsync("SELECT * FROM foodLog");
+  const lastEntry = data[data.length - 1];
```

Comment:
this logic fails if the database is uninitialised since your array would be empty

additionally, there's no point in selecting everything then taking the last - you should instead put that logic in the sql query itself via `LIMIT 1`

https://www.quora.com/How-do-you-access-the-last-row-of-an-SQLite-DB-in-Python